File: strongs/extract_strongs_indexes.py
import subprocess

# Run the diatheke command
result = subprocess.run(
    ["diatheke", "-b", "ASV", "-k", "Genesis 1:1", "-f", "OSIS"],
    capture_output=True,
    text=True,
    encoding="utf-8"
)

# Write the output to test.xml
with open("strongs_word_indexes.txt", "w", encoding="utf-8") as f:
    f.write(result.stdout)

File: strongs/extract_strongs_indexes.ts
import { spawnSync } from "child_process";
import fs from "fs-extra";
import path from "path";

async function run() {
    const outputPath = path.resolve(__dirname, "strongs_word_indexes.txt");
    const result = spawnSync("diatheke", ["-b", "KJV", "-k", "Genesis 1:1", "-f", "OSIS"], {
        encoding: "utf8",
    });

    if (result.error) {
        throw result.error;
    }

    await fs.outputFile(outputPath, result.stdout ?? "");
    console.log(`Wrote Strong's index output to ${path.relative(process.cwd(), outputPath)}`);
}

run().catch((error) => {
    console.error(error);
    process.exitCode = 1;
});
